cmd/zinject/zinject.c

```diff
@@ -896,6 +952,20 @@ main(int argc, char **argv)
 			usage();
 			libzfs_fini(g_zfs);
 			return (0);
+		case 'i':
+			errno = 0;
+			record.zi_timer = (int)strtol(optarg, &end, 10);
+			if (record.zi_timer <= 0 || *end != '\0') {
+				(void) fprintf(stderr, "invalid import delay "
+				    "'%s': must be a positive integer\n",
+				    optarg);
+				usage();
+				libzfs_fini(g_zfs);
+				return (1);
+			}
+			record.zi_timer = SEC2NSEC(record.zi_timer);
+			record.zi_cmd = ZINJECT_DELAY_IMPORT;
+			break;
```

Comment:
We have a 's' option which sets a duration. Can we combine the code so that we don't duplicate it?